ajax/ajax.md

```diff
@@ -38,52 +32,76 @@ Front-end code of interactive websites often needs to talk to backend servers to
 - Video walkthrough of lesson slides [AJAX](https://drive.google.com/file/d/1pLyMqvC-8cFroVTWMUBeG1_BRd4L-rL3/view?usp=sharing)
 
 
-### What is AJAX?
-
-AJAX stands for Asynchronous JavaScript and XML. It is a way for front-end code to talk to the backend by making *asynchronous* API calls.
+### What is AJAX ?
+AJAX stands for Asynchronous JavaScript and XML. It is a **technique** by which front-end get data from the backend by making *asynchronous* API calls.
+Because AJAX is *asynchronous*, you must pass in a callback function to handle the received data. Making an AJAX request without any callback function will not have any effect on your application.
 
-What is *asynchronous*?  You can think of this as a sequence of activities that respond to one another, but not in a continuous stream of time. Synchronous activities happen in a continuous, uninterrupted stream of time. A phone call, for example, is a synchronous method of communication, because two people are on the phone at the same time having a conversation. Email, on the other hand, is asynchronous communication. Even though a single conversation will happen across a thread of emails, each email by itself has a gap of time after the preceding email. On top of this, you never know when the next email response will arrive. This is the same thing that happens with asynchronous server calls. The front end will send a request to the backend, there is an indeterminable period of time that passes, then the server sends a response.
+### Full Load Page V/s AJAX
+> Lets say there is an input box and a button. Onclick of button the page shows some data say *hello world* send by the *server*. When the button is clicked -  
+#### Full load Application Flow
+1. The browser sends a request to a server designated by the URL. 
+2. The server does some processing, probably queries a database, then responds to the browser by sending back HTML and JavaScript
+3. The browser renders that HTML and JavaScript into a webpage.
 
-What is *XML*?  XML is a type of data interchange format, which means that it's a standard format that computers use to send data to one another. It stands for eXtensible Markup Language. While XML was commonly used when AJAX began, nowadays JSON has taken over as the preferred data format. This is because JSON is native to JavaScript. If your application was using XML, the application would have to take the extra step of converting XML into something that JavaScript could read and manipulate.
+#### AJAX Flow
+1. The client sends a request to the server.
+2. The server does some processing and returns the corresponding data (usually in JSON format).
+3. When the request gets completed the callback function is executed in which data is received. That data is shown on frontend through JavaScript in any format you want.
```

Comment:
```suggestion
3. When the request gets completed, the callback function is executed with the new data. The callback function could do anything you like; for example, it could use JS to add it to the HTML page.
```